### DISCUSSION TITLE
Tracking transaction state

### DISCUSSION THREAD

**Discussion by MartinGerritsen** [wevm/wagmi#758] [Q&A] [answered]
Hello! Loving your work, thank you so much for it!
I have a question; is it possible, using Wagmi, to track a transaction state? 
I know I receive the tx hash with which I could "do it manually", but I have been going through the documentation and I didn't find anything there, at least as far as I understood.
Hence the question, before creating a custom tracker :) 
Thanks!

**Comment by tmm** [comment, marked as answer]
Check out [`useWaitForTransaction`](https://wagmi.sh/docs/hooks/useWaitForTransaction). You can pass a transaction hash and it will update as the transaction is mining.
